### PR TITLE
Fix agent information when an upgrade or re-scan single agent is triggered

### LINKED ISSUE #23421: No agent name in vulnerability indexed data after upgrade
## Description
This issue aims to solve a problem when a database upgrade event is triggered. This occurs because, during the event context population, the agent_name field is not populated.

We need to get this information from the global database.

https://github.com/wazuh/wazuh/blob/de414d0991fa5e6090e8bdd125de2db404a17efb/src/wazuh_db/wdb_upgrade.c#L88

The effect of this is that the name of the agent is not indexed, something key for the user.

Is desired to have other missing fields (agent_ip and agent_version)

### PR DESCRIPTION
**Pull request by Dwordcito**
|Related issue|
|---|
|Closes #23421 |

This issue aims to solve a problem that caused certain information not to be indexed when an upgrade was performed or when the manager was rescanned after a policy change it.

### CODE REVIEW

**Review by GabrielEValenzuela — DISMISSED** [review]
LGTM!

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp [hunk 1]
```diff
@@ -44,8 +46,29 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
...
+            logDebug2(WM_VULNSCAN_LOGTAG, "Error executing query to fetch global agent data. Reason: %s.", e.what());
+            throw WdbDataException("Error executing query to fetch global agent data", data->agentId().data());
+        }
+        // LCOV_EXCL_STOP
+        // Agent in the first element of the array.
+        const auto& agent = response.back();
         data->m_agents.push_back(
-            {data->agentId().data(), data->agentName().data(), data->agentVersion().data(), data->agentIp().data()});
+            {data->agentId().data(), agent.at("name"), Utils::leftTrim(agent.at("version"), "Wazuh "), agent.at("ip")});
+
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp:71]: The response will be an empty array if no information is retrieved from an agent (eg, DB doesn't exist).

Accessing an empty array with back is undefined behavior ([see example](https://json.nlohmann.me/api/basic_json/back/#examples))

We can handle this by verifying the response is not empty

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp [hunk 2]
```diff
@@ -44,8 +46,33 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
...
+                    .globalGetCommand(std::string("agent-info ") + data->agentId().data())
+                    .build(),
+                response);
+        }
+        // LCOV_EXCL_START
+        catch (const std::exception& e)
+        {
+            logDebug2(WM_VULNSCAN_LOGTAG, "Error executing query to fetch global agent data. Reason: %s.", e.what());
+            throw WdbDataException("Error executing query to fetch global agent data", data->agentId().data());
+        }
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp:65]: We can add coverage to this with the following test

```C++
TEST_F(BuildSingleAgentListContextTest, ExceptionOnDB)
{
    spSocketDBWrapperMock = std::make_shared<MockSocketDBWrapper>();

    EXPECT_CALL(*spSocketDBWrapperMock, query(testing::_, testing::_))
        .Times(1)
        .WillOnce(testing::Throw(SocketDbWrapperException("Error on DB")));

    const auto jsonData = R"(
        {
            "agent_info": {
                "agent_id":"001"
            },
            "action":"upgradeAgentDB"
        })"_json;

    std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*, const nlohmann::json*>
        data = &jsonData;

    auto singleAgentContext = std::make_shared<
        TBuildSingleAgentListInfoContext<TScanContext<TrampolineOsDataCache>, TrampolineSocketDBWrapper>>();

    auto scanContext = std::make_shared<TScanContext<TrampolineOsDataCache>>(data);

    // Context is not used
    EXPECT_THROW(singleAgentContext->handleRequest(scanContext), WdbDataException);

    spSocketDBWrapperMock.reset();
}
```

**Review by sebasfalcone — DISMISSED** [review]
LGTM!

Just one detail to improve coverage

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp (hunk 2 above)
Comment (Dwordcito) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp:65]: yes, sure

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp [hunk 3]
```diff
@@ -44,8 +46,33 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
...
+        // LCOV_EXCL_STOP
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp:66]: No need for the lcov flags
```suggestion
        catch (const std::exception& e)
        {
            logDebug2(WM_VULNSCAN_LOGTAG, "Error executing query to fetch global agent data. Reason: %s.", e.what());
            throw WdbDataException("Error executing query to fetch global agent data", data->agentId().data());
        }
```

**Review by sebasfalcone — APPROVED** [review]
Agora sim! LGTM

**Pull request merged**